gnomad/utils/annotations.py

```diff
@@ -1519,3 +1529,308 @@ def _create_group_dicts(
+    :param label_version: String listing the version of the HT being used. Example: "3.1.2" .
+    :param coverage_ht: Hail Table containing coverage statistics, with mean depth stored in "mean" annotation. If None, omit coverage in return.
+    :param ancestry_groups: List of strings of shortened names of genetic ancestry groups to return results for. Example: ['afr','fin','nfe'] .
+    :param ancestry_groups_dict: Dict mapping shortened genetic ancestry group names to full names. Example: {'afr':'African/African American'} .
+    :param by_sex: Boolean to include breakdown of ancestry groups by inferred sex (XX and XY) as well.
+    :param vrs_only: Boolean to return only the VRS information and no general frequency information. Default is False.
+    :return: Dictionary containing VRS information (and frequency information split by ancestry groups and sex if desired) for the specified variant.
+    """
+    # Throw warnings if contradictory arguments passed.
+    if by_sex and not ancestry_groups:
+        logger.warning(
+            "Splitting whole database by sex is not yet supported. If using 'by_sex',"
+            " please also specify 'ancestry_groups' to stratify by."
+        )
+
+    ht = ht.annotate(
+        gnomad_id=hl.format(
+            "%s-%s-%s-%s",
+            ht.locus.contig,
+            ht.locus.position,
+            ht.alleles[0],
+            ht.alleles[1],
+        )
+    )
+
+    # Define function to return a frequency report dictionary for a given group
+    def _create_group_dicts(
+        group_index: int,
+        group_id: str,
+        group_label: str,
+        group_sex: str = None,
+    ) -> dict:
+        """
+        Return a dictionary for the frequency information of a given variant for a given subpopulation.
+
+        :param group_index: Index of frequency within the 'freq' annotation containing the desired group.
+        :param group_id: String containing variant, genetic ancestry group, and sex (if requested). Example: "chr19-41094895-C-T.afr.XX".
+        :param group_label: String containing the full name of genetic ancestry group requested. Example: "African/African American".
+        :param group_sex: String indicating the sex of the group. Example: "XX", or "XY".
+        :return: Dictionary containing VRS information (and genetic ancestry group if desired) for specified variant.
+        """
+        # Obtain frequency information for the specified variant
+        group_freq = ht.freq[group_index]
+
+        # Cohort characteristics
+        characteristics = []
+        characteristics.append({"name": "genetic ancestry", "value": group_label})
+        if group_sex is not None:
+            characteristics.append({"name": "biological sex", "value": group_sex})
+
+        # Dictionary to be returned containing information for a specified group
+        freq_record = {
+            "id": hl.format("%s.%s", ht.gnomad_id, group_id.upper()),
```

Comment:
why use hl.format here rather than fstrings, such as "f"{variant}.{group_id.upper()}"?

Reply:
not sure, but I vote for fstrings, so will add

Reply:
It is using `hl.format` because the `ht.gnomad_id` is a hail `StringExpression`, not a `str`